libs/openFrameworks/utils/ofThreadChannel.h

```diff
@@ -1,28 +1,97 @@
 #pragma once
+
+
 #include <mutex>
 #include <queue>
 #include <condition_variable>
 #include "ofUtils.h"
 
-/// Communication channel between different threads
-/// allows for multithreaded programming without using
-/// mutexes or other synchronization structures.
-/// Instead of using some shared memory send a message
-/// through a channel whenever one thread needs to send
-/// some data to another.
-/// Usually a second channel can be used to send the
-/// processed data back to the first thread.
+
+/// \brief Safely send data between threads without additional synchronization.
+///
+/// ofThreadChannel is a powerful class that makes it easy to safely pass data
+/// between threads. When data is sent form one thread to another via the
+/// ofThreadChannel, the user does not need to use additonal mutexes, condition
+/// variables, critical sections, semaphores, monitors or other synchronization
+/// primitives or techniques. Data sent with ofThreadChannel is synchronized
```

Comment:
i don't think it's necessary to mention every possible sync primitive here :) doesn't add to the explanation and it might be confusing i think